python/ray/util/sgd/torch/worker_group.py

```diff
@@ -135,16 +137,47 @@ def __init__(self, max_workers, params, dist_params, initialization_hook,
         # The last time when this worker group was resized.
         self._last_resize = float("-inf")
 
+        # This is set only if a placement group is created by the worker group.
+        self._worker_placement_group = None
+
+    def _create_placement_group(self, num_workers):
+        """Creates a placement group for the workers.
+
+        If this worker is already in a placement group then a new one will
+        not be created. This is primarily for when Tune is the upstream and
+        will allocate resources for SGD workers.
+
+        If this worker is not in a placement group, a new one will be created
+        and set. The placement group will have a single bundle for each worker
+        and use the SPREAD strategy for an even distribution.
+        """
+        pg = get_current_placement_group()
+        if pg is None:
+            bundle = {
+                "CPU": self._num_cpus_per_worker,
+                "GPU": int(self._use_gpu)
+            }
+            bundles = [bundle] * num_workers
+            pg = ray.util.placement_group(bundles, strategy="SPREAD")
+            logger.debug("Waiting for placement group to start.")
+            ray.get(pg.ready(), timeout=self._timeout_s)
```

Comment:
Instead of `ray.get` here could you do something like this instead: https://github.com/ray-project/xgboost_ray/blob/master/xgboost_ray/main.py#L758-L768

This way we can raise an error message that's more interpretable to the user.